pkg/util/strategicpatch/patch.go

```diff
@@ -963,13 +972,170 @@ func HasConflicts(left, right interface{}) (bool, error) {
 	}
 }
 
+// MergingMapsHaveConflicts returns true if the left and right JSON interface
```

Comment:
This method supports merging lists (i.e., strategic merge path semantics), while the other one doesn't (i.e., json merge patch semantics). Will update to make the distinction clear.